modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/util/DDMFormValidationUtil.java

```
@@ -0,0 +1,123 @@
+
+/**
+ * @author Vamshi Krishna
+ */
+public class DDMFormValidationUtil {
+
+	public static boolean isValidDDMFormValues(FileVersion fileVersion) {
+		try {
+			Map<String, com.liferay.dynamic.data.mapping.kernel.DDMFormValues>
+				ddmFormValuesMap = new HashMap<>();
+
+			List<DDMStructure> ddmStructures;
+
+			if (fileVersion.getFileEntryId() > 0) {
+				DLFileVersion dlFileVersion =
+					DLFileVersionLocalServiceUtil.getFileVersion(
+						fileVersion.getFileEntryId(), fileVersion.getVersion());
+
+				DLFileEntryType dlFileEntryType =
+					DLFileEntryTypeLocalServiceUtil.getFileEntryType(
+						dlFileVersion.getFileEntryTypeId());
+
+				ddmStructures = dlFileEntryType.getDDMStructures();
+			}
+			else {
+				ddmStructures = DDMStructureManagerUtil.getClassStructures(
+					fileVersion.getCompanyId(),
+					ClassNameLocalServiceUtil.getClassNameId(
+						DLFileEntryMetadata.class));
+			}
+
+			for (DDMStructure ddmStructure : ddmStructures) {
+				DLFileEntryMetadata dlFileEntryMetadata =
+					DLFileEntryMetadataLocalServiceUtil.fetchFileEntryMetadata(
+						ddmStructure.getStructureId(),
+						fileVersion.getFileVersionId());
+
+				if (dlFileEntryMetadata == null) {
+					continue;
+				}
+
+				ddmFormValuesMap.put(
+					ddmStructure.getStructureKey(),
+					StorageEngineManagerUtil.getDDMFormValues(
+						dlFileEntryMetadata.getDDMStorageId()));
+			}
+
+			for (DDMStructure ddmStructure : ddmStructures) {
+				com.liferay.dynamic.data.mapping.kernel.DDMFormValues
+					ddmFormValues = ddmFormValuesMap.get(
+						ddmStructure.getStructureKey());
+
+				DDMFormValues translatedDDMFormValues =
+					DDMBeanTranslatorUtil.translate(ddmFormValues);
+
+				if (ddmFormValues != null) {
+					validate(translatedDDMFormValues);
+				}
+			}
+		}
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(exception);
+			}
+
+			if (exception instanceof DDMFormValuesValidationException) {
+				return false;
+			}
+		}
```

Avoid catching `Exception` if possible (it is too broad). Instead, catch the expected errors and let the others bubble up.
```
catch (DDMFormValuesValidationException | OtherException | YetAnotherOneException exception) {
    if (_log.isDebugEnabled()) {
        ...
    }

    return true;
}
```

---

Fixed, Thank You